src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,176 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item, Node<T> next, Node<T> prev) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void checkIndexAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private void checkIndexSet(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private void checkIndexRemove(int index) {
+        if (index < 0 || index == size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private Node iterateMyLinkedList(int index) {
+        int count = 0;
+        if (index <= size / 2) {
+            Node currentNode = first;
+            while (count != index) {
+                currentNode = currentNode.next;
+                count++;
+            }
+            return currentNode;
+        }
+        if (index > size / 2) {
+            Node currentNode = last;
+            while (count != size - index - 1) {
+                currentNode = currentNode.prev;
+                count++;
+            }
+            return currentNode;
+        }
+        return null;
+    }
+
+    private void unlink(Node nodeToUnlink) {
+        if (nodeToUnlink == first && nodeToUnlink == last) {
+            first = null;
+            last = null;
+            return;
+        }
+        if (nodeToUnlink == first) {
+            nodeToUnlink.next.prev = null;
+            first = nodeToUnlink.next;
+            return;
+        }
+        if (nodeToUnlink == last) {
+            nodeToUnlink.prev.next = null;
+            last = nodeToUnlink.prev;
+            return;
+        }
+        nodeToUnlink.prev.next = nodeToUnlink.next;
+        nodeToUnlink.next.prev = nodeToUnlink.prev;
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node node = new Node<>(value, null, null);
+            first = node;
+            last = node;
+        }
+        if (size > 0) {
+            Node node = new Node<>(value, null, null);
+            Node currentNode = last;
+            currentNode.next = node;
+            node.prev = currentNode;
+            last = node;
```

Comment:
think about hot to simplify this code and avoid duplication

Reply:
I moved Node creation out from if statements directly to method, moved "last = node" to the end, as we always add to the end.
I can't think of other simplifications, it looks pretty simple to me already. If I am missing something obvious - do not hesitate to point that to me, that's interesting